sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayFunctionTests.java

```diff
@@ -0,0 +1,616 @@
+                (1 row)
+                """
+        );
+    }
+
+    @Test @Ignore("ICE: compilation failure while optimizing")
+    public void testArrayRepeat3() {
+        this.qs("""
+                SELECT array_repeat('a', 6);
+                 array_repeat
+                --------------
+                 { a, a, a, a, a, a}
+                (1 row)
+                """
+        );
+    }
+
+    @Test
+    public void testArrayRemove() {
+        this.qs("""
+                SELECT array_remove(ARRAY [2, 2, 6, 6, 8, 2], 2);
+                 array_remove
+                --------------
+                 {6, 6, 8}
+                (1 row)
+                
+                SELECT array_remove(ARRAY [1, 2, 3], null);
+                 array_remove
+                --------------
+                 NULL
+                (1 row)
+                
+                SELECT array_remove(ARRAY [null, 2, 2, 6, 6, 8, 2], 2);
+                 array_remove
+                --------------
+                 {null, 6, 6, 8}
+                (1 row)
+                
+                SELECT array_remove(ARRAY [null, 2, 2, 6, 6, 8, 2], null);
+                 array_remove
+                --------------
+                 NULL
+                (1 row)
+                
+                SELECT array_remove(ARRAY [2, 2, 6, 6, 8, 2], elem) FROM (SELECT elem FROM UNNEST(ARRAY [2, 6, 8]) as elem);
+                 array_remove
+                --------------
+                 {6, 6, 8}
+                 {2, 2, 8, 2}
+                 {2, 2, 6, 6, 2}
+                (3 rows)
+                
+                SELECT array_remove(ARRAY [2, 2, 6, 6, 8, 2], elem) FROM (SELECT elem FROM UNNEST(ARRAY [2, 6, 8, null]) as elem);
+                 array_remove
+                --------------
+                 {6, 6, 8}
+                 {2, 2, 8, 2}
+                 {2, 2, 6, 6, 2}
+                 NULL
+                (4 rows)
+                
+                """
+        );
+    }
+
+    @Test @Ignore("TBD: Should we support implicitly casting here? Snowflake thinks its fine as long as types are comparable")
+    public void testArrayFnsDiffTypes() {
+        this.qs("""
+                SELECT array_position(ARRAY [1, 2, 4, 6], 1e0);
+                 array_position
+                ----------------
+                 1
+                (1 row)
+                
+                SELECT array_position(ARRAY [1, 2, 4, 6], 1e0);
+                 array_position
+                ----------------
+                 1
+                (1 row)
+                
+                SELECT array_position(ARRAY [NULL], 1);
+                 array_position
+                ----------------
+                 0
+                (1 row)
+                
+                SELECT array_contains(ARRAY [NULL], 1);
+                 array_contains
+                ----------------
+                 false
+                (1 row)
+                """
+        );
```

Comment:
Here should we try to implicitly cast the argument to the element type of the array? 

But if we do that, arrays like `ARRAY [NULL]` are very awkward to deal with. For them, do we cast the element to the argument type?